src/ifaddrs.rs

```diff
@@ -0,0 +1,180 @@
+//! Query network interface addresses
+//!
+//! Uses the Linux and/or BSD specific function getifaddrs to query the list
+//! of interfaces and their associated addresses.
+
+use std::ffi;
+use std::iter::Iterator;
+use std::mem;
+use std::option::Option;
+
+use libc;
+use {Result, Error};
+
+use sys::socket::InetAddr;
+
+#[cfg(any(target_os = "linux", target_os = "emscripten", target_os = "fuchsia"))]
+mod flags {
+    use libc;
+    libc_bitflags!(
+        /// Flags for `InterfaceAddress` entries returned by `getifaddrs`
+        pub flags InterfaceFlags: libc::c_int {
+            IFF_UP,
+            IFF_BROADCAST,
+            IFF_DEBUG,
+            IFF_LOOPBACK,
+            IFF_POINTOPOINT,
+            IFF_NOTRAILERS,
+            IFF_RUNNING,
+            IFF_NOARP,
+            IFF_PROMISC,
+            IFF_ALLMULTI,
+            IFF_MASTER,
+            IFF_SLAVE,
+            IFF_MULTICAST,
+            IFF_PORTSEL,
+            IFF_AUTOMEDIA,
+            IFF_DYNAMIC,
+        }
+    );
+}
+
+#[cfg(not(any(target_os = "linux", target_os = "emscripten", target_os = "fuchsia")))]
+mod flags {
+    use libc;
+    /// Flags for `InterfaceAddress` entries returned by `getifaddrs`
+    libc_bitflags!(
+        pub flags InterfaceFlags: libc::c_int {
+            IFF_LOOPBACK,
+        }
+    );
+}
+
+pub use self::flags::*;
+
+/// Describes a single address for an interface as returned by `getifaddrs`.
+pub struct InterfaceAddress {
+    /// name of the interface
+    pub interface_name: String,
+    /// flags from `SIOCGIFFLAGS`
+    pub flags: InterfaceFlags,
+    /// address of interface
+    pub address: Option<InetAddr>,
+    /// netmask of interface
+    pub netmask: Option<InetAddr>,
+    /// broadcast address of interface, if applicable
+    pub broadcast: Option<InetAddr>,
+    /// point-to-point destination address
+    pub destination: Option<InetAddr>,
+}
+
+impl InterfaceAddress {
+    /// Create an `InterfaceAddress` from the libc struct.
+    fn from_libc_ifaddrs(info: &libc::ifaddrs) -> InterfaceAddress {
```

Comment:
Exported? It's not public and used internally, so I fear I don't understand what you're asking for.